clients/src/test/java/org/apache/kafka/clients/admin/NewTopicTest.java

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.message.CreateTopicsRequestData;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.kafka.common.requests.CreateTopicsRequest.NO_NUM_PARTITIONS;
+import static org.apache.kafka.common.requests.CreateTopicsRequest.NO_REPLICATION_FACTOR;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+public class NewTopicTest {
+
+    public static final String TEST_TOPIC = "testtopic";
+    public static final int NUM_PARTITIONS = 3;
+    public static final int REPLICATION_FACTOR = 1;
+    public static final String CLEANUP_POLICY_CONFIG_KEY = "cleanup.policy";
+    public static final String CLEANUP_POLICY_CONFIG_VALUE = "compact";
+    public static final List<Integer> BROKER_IDS = Arrays.asList(1, 2);
+
+    @Test
+    public void testConstructorWithPartitionsAndReplicationFactor() {
+        NewTopic topic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, (short) REPLICATION_FACTOR);
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithOptionalValues() {
+        Optional<Integer> numPartitions = Optional.empty();
+        Optional<Short> replicationFactor = Optional.empty();
+        NewTopic topic = new NewTopic(TEST_TOPIC, numPartitions, replicationFactor);
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NO_NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithReplicasAssignments() {
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(0, BROKER_IDS);
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        assertEquals(TEST_TOPIC, newTopic.name());
+        assertEquals(NO_NUM_PARTITIONS, newTopic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, newTopic.replicationFactor());
+        assertEquals(replicasAssignments, newTopic.replicasAssignments());
+    }
+
+    @Test
+    public void testConfigs() {
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, (short) REPLICATION_FACTOR);
+        Map<String, String> configs = new HashMap<>();
```

Comment:
Could you verify that `newTopic.configs()` will return null if it has no configs

Reply:
Indeed, added one.